### ISSUE TITLE
[Spell] Wings of the Damned

### ISSUE THREAD

**Issue by Moriquendu** [TrinityCore/TrinityCore#4169]
Hello,

There seems to be a slight bug, with handling of spell auras 305 (SPELL_AURA_MOD_MINIMUM_SPEED)

Right now, the code states:

``` c++
if (float minSpeedMod = (float)GetMaxPositiveAuraModifier(SPELL_AURA_MOD_MINIMUM_SPEED))
{
        float min_speed = minSpeedMod / 100.0f;
        if (speed < min_speed)
        speed = min_speed;
}
```

The point is, the value of the "Minimum speed" is taken as an absolute value, instead of as value relative to the unit's base speed rate.

ie. If we have two units,
unit A with base speed rate 1.0,
unit B with base speed rate 0.6, 

And we apply a -70% slow to both of them,

unit A will be slowed only by 50% ( down to 0.5 ), which is correct, but 
unit B will be slowed down to 0.5, instead of 50% of 0.6 ( 50% of base speed rate ).

How to quickly test in practice:
Place unit A and unit B, apply a slow debuff on both of them ( ie. Cone of Cold ), observe how they both run at same speed.

Note: I'm not 100% about correct expected behaviour, couldn't verify for or against it.

Thank you.
##

**Comment by Keader** [comment]
This still valid on 166e66cc06d418b94e14abb30dea6a76ba9edce2

**Comment by sirikfoll** [comment]
The correction proposed works fine, as its supposed to only affect spells with SPELL_AURA_MOD_MINIMUM_SPEED,
it will affect the spells
48165
67116
69406
69416
74352
Also, seems that they are mostly unused
```diff
diff --git a/src/server/game/Entities/Unit/Unit.cpp b/src/server/game/Entities/Unit/Unit.cpp
index d02a22d..3b6e115 100644
--- a/src/server/game/Entities/Unit/Unit.cpp
+++ b/src/server/game/Entities/Unit/Unit.cpp
@@ -12386,7 +12386,8 @@ void Unit::UpdateSpeed(UnitMoveType mtype, bool forced)
 
     if (float minSpeedMod = (float)GetMaxPositiveAuraModifier(SPELL_AURA_MOD_MINIMUM_SPEED))
     {
-        float min_speed = minSpeedMod / 100.0f;
+        float baseSpeed = (GetTypeId() == TYPEID_UNIT ? ToCreature()->GetCreatureTemplate()->speed_walk : 1.0f);
+        float min_speed = (minSpeedMod / 100.0f) * baseSpeed;
         if (speed < min_speed)
             speed = min_speed;
     }

**Comment by Aokromes** [comment]
This bug is still valid?

**Comment by sirikfoll** [comment]
Yes

**Comment by Aokromes** [comment]
@Shauren https://github.com/TrinityCore/TrinityCore/issues/4169#issuecomment-138384179 is valid?

**Comment by Aokromes** [comment]
@sirikfoll can you create a PR for this?

**Comment by Shauren** [comment]
Why is it using walk speed?

**Comment by sirikfoll** [comment]
`speed_walk` was on the original suggestion, i'm currently using this with `speed_run`, but it will also need a Db change in order to properly work, which i think it's not correct:
```sql
UPDATE `creature_template` SET `speed_run`=0.642857 WHERE `entry` IN (36609,39120,39121,39122);
```
if the Db change is acceptable, just tell me and i open the PR ^^

**Comment by sirikfoll** [comment]
Suggested Fix
```diff
diff --git a/src/server/game/Entities/Unit/Unit.cpp b/src/server/game/Entities/Unit/Unit.cpp
index e004ef2..77ce20b 100644
--- a/src/server/game/Entities/Unit/Unit.cpp
+++ b/src/server/game/Entities/Unit/Unit.cpp
@@ -11965,7 +11965,10 @@ void Unit::UpdateSpeed(UnitMoveType mtype)
                     speed *= pOwner->GetSpeedRate(mtype) * mult; // pets derive speed from owner when not in combat
                 }
                 else
-                    speed *= ToCreature()->GetCreatureTemplate()->speed_run;    // at this point, MOVE_WALK is never reached
+                {
+                    float curSpeed = GetSpeedRate(MOVE_RUN);
+                    speed *= curSpeed > 0 ? curSpeed : ToCreature()->GetCreatureTemplate()->speed_run;    // at this point, MOVE_WALK is never reached
+                }
             }
 
             // Normalize speed by 191 aura SPELL_AURA_USE_NORMAL_MOVEMENT_SPEED if need
@@ -12004,7 +12007,7 @@ void Unit::UpdateSpeed(UnitMoveType mtype)
 
     if (float minSpeedMod = (float)GetMaxPositiveAuraModifier(SPELL_AURA_MOD_MINIMUM_SPEED))
     {
-        float min_speed = minSpeedMod / 100.0f;
+        float min_speed = (minSpeedMod / 100.0f) * GetSpeedRate(MOVE_RUN);
         if (speed < min_speed)
             speed = min_speed;
     }
diff --git a/src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp b/src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp
index de8d656..76f1c09 100644
--- a/src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp
+++ b/src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp
@@ -1449,7 +1449,7 @@ class npc_valkyr_shadowguard : public CreatureScript
                 _events.Reset();
                 me->SetReactState(REACT_PASSIVE);
                 DoCast(me, SPELL_WINGS_OF_THE_DAMNED, false);
-                me->SetSpeedRate(MOVE_FLIGHT, 0.642857f);
+                me->SetSpeedRate(MOVE_RUN, 0.642857f);
             }
 
             void IsSummonedBy(Unit* /*summoner*/) override

**Comment by tkrokli** [comment]
@sirikfoll : Do you have time to open a PR for your suggested fix? :-)

**Comment by Frytiks** [comment]
Personally, i changed all MOVE_FLIGHT to MOVE_RUN for all flying creatures.

**Comment by sirikfoll** [comment]
Closed by #18136

**Comment by Moriquendu** [comment]
Awesome, thanks ;)
